airbyte-integrations/connectors/source-mailchimp/source_mailchimp/streams.py

```diff
@@ -129,11 +134,30 @@ def stream_slices(
     ) -> Iterable[Optional[Mapping[str, Any]]]:
         slice_ = {}
         stream_state = stream_state or {}
-        cursor_value = stream_state.get(self.cursor_field)
+        cursor_value = self.get_filter_date(self.start_date, stream_state.get(self.cursor_field))
         if cursor_value:
             slice_[self.filter_field] = cursor_value
         yield slice_
 
+    @staticmethod
+    def get_filter_date(start_date: str, state_date: str) -> str:
+        """
+        Calculate the filter date to pass in the request parameters by comparing the start_date
+        with the value of state obtained from the stream_slice.
+        If only one value exists, use it by default. Otherwise, return None.
+        If no filter_date is provided, the API will fetch all available records.
+        """
+
+        start_date_parsed = pendulum.parse(start_date).to_iso8601_string() if start_date else None
+        state_date_parsed = pendulum.parse(state_date).to_iso8601_string() if state_date else None
+
+        if start_date_parsed and state_date_parsed:
+            return max(start_date_parsed, state_date_parsed)
+        elif state_date_parsed or start_date_parsed:
+            return state_date_parsed or start_date_parsed
+        else:
+            return None
```

Comment:
```suggestion
        # Return the maximum of the two dates if both are present, otherwise return whichever is present, or None
        if start_date_parsed or state_date_parsed:
                return max(filter(None, [start_date_parsed, state_date_parsed]), default=None)
```

Reply:
Updated